### PR TITLE
4 feature 4a card dealing system

### PR DESCRIPTION
**Pull request by Andy-Guo-21**
Card and Deck implementation done.

### CODE REVIEW

File: src/main/java/Card.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            rank = "jack";
+        } else if (this.getRank().equalsIgnoreCase("Q")) {
+            rank = "queen";
+        } else if (this.getRank().equalsIgnoreCase("K")) {
+            rank = "king";
+        } else {
+            rank = this.getRank().toString();
+        }
+
+        return "images/Playing Cards/" + rank + "_of_" + suit + ".png";
```
Comment (Jefferson-liu) [inline src/main/java/Card.java:163]: NEED TO WRITE TESTS

File: src/test/java/CardTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Test
+    void compareSuit() {
+    }
+
+    @Test
+    void compareRank() {
+    }
+}
```
Comment (Jefferson-liu) [inline src/test/java/CardTest.java:22]: No tests

File: src/test/java/DeckTest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    @Test
+    void isEmpty() {
+    }
+
+    @Test
+    void getCard() {
+    }
+}
```
Comment (Jefferson-liu) [inline src/test/java/DeckTest.java:18]: No tests

**Review by Jefferson-liu — APPROVED** [review]
No tests but good  code

**Review by enfz — APPROVED** [review]
Entity class implementations are good and sufficient to be usable for the other classes, would need more tests in the future to fully work through any bugs.

**Review by a-llison-lau — APPROVED** [review]
Can implement some tests to test the code. Javadocs are clear.

**Pull request merged**